geometry/WaitingArea.h

```diff
@@ -157,8 +157,9 @@ class WaitingArea : public Goal{
     void SetWaitingTime(int waitingTime);
 
     /**
-     * Returns the ID of the next goal based on the probability
-     * @return ID of the next goal
+     * Returns the ID of the next goal based on the probability. Only
+     * considers open goals.
+     * @return ID of the next goal or own ID if no open goal available
```

Comment:
Probably too much for this PR, but for functions returning "nothing" like no goal found we should use [std::optional](https://en.cppreference.com/w/cpp/utility/optional).